auth/src/main/java/org/aerogear/mobile/auth/ConnectionBuilderForTesting.java

```diff
@@ -0,0 +1,112 @@
+package org.aerogear.mobile.auth;
+
+import android.annotation.SuppressLint;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import net.openid.appauth.Preconditions;
+import net.openid.appauth.connectivity.ConnectionBuilder;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.X509Certificate;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+/**
+ * An example implementation of {@link ConnectionBuilder} that permits connecting to http
+ * links, and ignores certificates for https connections. *THIS SHOULD NOT BE USED IN PRODUCTION
+ * CODE*. It is intended to facilitate easier testing of AppAuth against development servers
+ * only.
+ */
+public class ConnectionBuilderForTesting implements ConnectionBuilder {
```

Comment:
Is there anything significantly different that we are doing that is different to the [AppAuth test class](https://github.com/openid/AppAuth-Android/blob/master/app/java/net/openid/appauthdemo/ConnectionBuilderForTesting.java). Do we still need this or is of benefit? It looks like this a test for self-signed certs or HTTP connections only, which AppAuth has features for, but we likely won't be using this as it will be part of our own cert pinning implementation instead and might get some more relevant tests there? WDYT?

TL;DR I think this is a test for a feature we aren't currently or might not use IMO.

Reply:
I agree. We can remove this once we have the certificate pinning implementation in place. For now we do have to support self-signed certs as most developers will run the services with self-signed cert.